lib/ap/plan9/malloc.c

```diff
@@ -37,7 +37,7 @@ struct Arena
 };
 static Arena arena;
 
-#define datoff		((int)((Bucket*)0)->data)
+#define datoff		((long)((Bucket*)0)->data)
 #define nil		((void*)0)
```

Comment:
This being a standard posix world, I would consider getting rid of all uses of nil. NULL is the standard. This just adds more confusing I think.